cipher/src/stream_core.rs

```diff
@@ -35,7 +38,7 @@ pub trait StreamClosure: BlockSizeUser {
 }
 
 /// Block-level synchronous stream ciphers.
-pub trait StreamCipherCore: BlockSizeUser + Sized {
+pub trait StreamCipherCore: BlockSizeUser + ParBlocksSizeUser + Sized {
```

Comment:
As I wrote in the issue, this approach is incorrect. `StreamCipherCore` does not "know" at compile time the number of blocks which can be generated in parallel at runtime. It's known only by backends which can be selected using runtime target feature detection. For example, `chacha20` has SSE2 backend which generates only [one block](https://github.com/RustCrypto/stream-ciphers/blob/master/chacha20/src/backends/sse2.rs#L45-L47) and AVX2 which generates [4 blocks](https://github.com/RustCrypto/stream-ciphers/blob/master/chacha20/src/backends/avx2.rs#L59-L61). By default on x86 targets the crate uses runtime autodetection to select between those.

Reply:
The approach is not ideal, but it would still work if the x86 `ParBlocksSize` for `ChaChaCore` was set to `U4`.

Is there a better approach for choosing the size at runtime? The last time I tried to pick a variable-sized `[T; N]` at runtime, the compiler was yelling at me because it did not know `N` at compile time. Is this possible with the `array` crate, or something similar?